MetaMask/metamask-extension
# Ability to buy tokens with Moonpay

## Explanation

**This extends this already merged PR**: https://github.com/MetaMask/metamask-extension/pull/15551
and adds Moonpay as an extra onramp provider

Current State: Only native currencies can be bought through Moonpay using the buy button
After Change: Ability to purchase tokens defined in BUYABLE_CHAINS_MAP using Moonpay



## More Information

See ONRAMP-189 and ONRAMP-190



## Screenshots/Screencaps



### Before
<img width="354" alt="Screenshot 2022-09-21 at 16 12 46" src="https://user-images.githubusercontent.com/8658960/191527879-38353149-f7b4-4a40-a6db-e7d9497a5e80.png">



### After
<img width="355" alt="Screenshot 2022-09-21 at 16 12 32" src="https://user-images.githubusercontent.com/8658960/191527910-9d3fa242-616e-446a-988d-20f866931745.png">



## Manual Testing Steps

These changes apply only to Ethereum mainnet

Import a token
If token is on the Moonpay lists it should display a "Buy" button

Moonpay lists:

[Mainnet](https://github.com/nikoferro/metamask-extension/commit/15959485c202fa0dac3fd1b4f0b0d934a44c80a4#diff-4678018988a2b2341bb3addb3e8ae051c9fe7778e7b8ed3fdd7834a3f5912dd2R683)
[BSC](https://github.com/nikoferro/metamask-extension/commit/15959485c202fa0dac3fd1b4f0b0d934a44c80a4#diff-4678018988a2b2341bb3addb3e8ae051c9fe7778e7b8ed3fdd7834a3f5912dd2R828):
[Polygon](https://github.com/nikoferro/metamask-extension/commit/15959485c202fa0dac3fd1b4f0b0d934a44c80a4#diff-4678018988a2b2341bb3addb3e8ae051c9fe7778e7b8ed3fdd7834a3f5912dd2R845)
[Avalanche C-Chain](https://github.com/nikoferro/metamask-extension/commit/15959485c202fa0dac3fd1b4f0b0d934a44c80a4#diff-4678018988a2b2341bb3addb3e8ae051c9fe7778e7b8ed3fdd7834a3f5912dd2R859)
[Celo](https://github.com/nikoferro/metamask-extension/commit/15959485c202fa0dac3fd1b4f0b0d934a44c80a4#diff-4678018988a2b2341bb3addb3e8ae051c9fe7778e7b8ed3fdd7834a3f5912dd2R880)

Depending on which list it shows, it should give you the option to buy the token on Moonpay
Once you click on the provider's "Continue to" button, a window on said provider should open, and the token to buy should already be preselected

## Pre-Merge Checklist


### + If there are functional changes:
